similar_dev_search/model/developer.py

```diff
@@ -33,10 +39,51 @@ async def get_stargazes(self, asyncio_client: httpx.AsyncClient = None) -> List[
 
         return starred_repos
 
+    async def get_languages(self, asyncio_client: httpx.AsyncClient = None) -> defaultdict[int]:
+        """
+        Gets dict of languages used by the developer
+        :param asyncio_client: asyncio client to perform requests from
+        :return: dict with language key and number of occurrences value
+        """
+        if self.languages is not None:
+            return self.languages
+        languages, _ = await self.get_languages_variables(asyncio_client)
+        return languages
+
+    async def get_variables(self, asyncio_client: httpx.AsyncClient = None) -> defaultdict[int]:
+        """
+        Gets dict of variables used by the developer
+        :param asyncio_client:  asyncio client to perform requests from
+        :return: dict with variable key and number of occurrences value
+        """
+        if self.variables is not None:
+            return self.variables
+        _, variables = await self.get_languages_variables(asyncio_client)
+        return variables
+
+    async def get_languages_variables(self, asyncio_client: httpx.AsyncClient = None) \
+            -> Tuple[defaultdict[int], defaultdict[int]]:
+        """
+        Gets both languages and variables that developer uses
+        :param asyncio_client:  asyncio client to perform requests from
+        :return: dict of languages, dict of variables
+        """
+        if self.languages is not None and self.variables is not None:
+            return self.languages, self.variables
+        self.languages = defaultdict(int)
+        self.variables = defaultdict(int)
+        repos = await self.get_repos()
+
+        for i in range(min(repos_limit, len(repos))):
```

Comment:
wrong logic of work.

How it should work - on previous step you had to collect top repositories like this:
* start with some repository - get all stargazers
* for each stargazer find all starred repositories - 
    *so you will have - `dev2starred_repos = {developer: [list of starred repositories]}`
* then you count how many times each repository appeared among stargazers
```python
repos2cnt = Counter()
for repos in dev2starred_repos.values():
     repos2cnt.update(repos)
```
* after this step you clone top-10/20 repositories from these top repositories
Logic for 1 repository:
* iterate through commits of repository
* for each commit with 1 parent (so it's not merge commit) 
    * find author of commit
    * find how many lines in files were modified by this author - store this information (it should be added together)
    * if modified file was not parsed (or language of file is unknown) - parse file, keep list of identifiers and language for this file
* after processing all repositories you will have several important dataset
    * `{developer: {repo: {filepath: {added: 100, deleted: 100}}}`
    * {repo: {filepath: {language: python, identifiers: [word1, word2...]}}}
 
So you can use this information to create 2 vectors for each developer
* language vector - like {python: 100, bash: 12, java: 67,...}
* identifiers vector - like {torch: 200, matmul: 1122, sum: 67,...}
And you can use these vectors to calculate similarity among developers